posthog/models/person_overrides/sql.py

```diff
@@ -141,3 +141,26 @@
     ON CLUSTER '{CLICKHOUSE_CLUSTER}'
     SYNC
 """
+
+# ClickHouse dictionaries allow us to JOIN events with their new override_person_ids (if any).
+PERSON_OVERRIDES_CREATE_DICTIONARY_SQL = f"""
+    CREATE OR REPLACE DICTIONARY IF NOT EXISTS `{CLICKHOUSE_DATABASE}`.`person_overrides_dict`
+    ON CLUSTER '{CLICKHOUSE_CLUSTER}' (
+        team_id INT,
+        old_person_id UUID,
+        override_person_id UUID
+    )
+    PRIMARY KEY team_id, old_person_id
+    SOURCE(CLICKHOUSE(TABLE 'person_overrides' DB '{CLICKHOUSE_DATABASE}'))
```

Comment:
We'll need to take the latest version of the record (by team_id & old_person_id), for that we'll need to use the optional `query` param in the source object [[see doc]](https://clickhouse.com/docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-sources/#clickhouse). Can you please look into that? I unfortunately don't have enough CH-fu to help here, but I know this is common query pattern that we have (for example in the `person` table). Harry could help if needed.